File: createNewIndex.ts
import { Pinecone } from "@pinecone-database/pinecone";

export default async (
  client: Pinecone,
  indexName: string
) => {
  const { indexes } = await client.listIndexes();

  for (const index of indexes) {
    console.log(`"${index.name}" will be deleted.`);
    await client.deleteIndex(index.name);
  }

  console.log(`Creating "${indexName}"...`);

  const createClient = await client.createIndex({
    name: indexName,
    dimension: 1536,
    metric: "cosine",
    spec: {
      serverless: {
        cloud: 'aws', region: 'us-east-1'
      }
    }
  });

  console.log(`Created with client:`, createClient);

  await new Promise((resolve) => setTimeout(resolve, 10000));
};

File: python/create_new_index.py
import asyncio
from pinecone import Pinecone


async def create_new_index(client: Pinecone, index_name: str):
    indexes = client.list_indexes()

    for index in indexes:
        print(f'"{index.name}" will be deleted.')
        client.delete_index(index.name)

    print(f'Creating "{index_name}"...')

    client.create_index(
        name=index_name,
        dimension=1536,
        metric="cosine",
        spec={"serverless": {"cloud": "aws", "region": "us-east-1"}},
    )

    await asyncio.sleep(10)
